src/Components/OATGraphViewer/Internal/OATGraphCustomEdge.tsx

```diff
@@ -464,28 +469,26 @@ const OATGraphCustomEdge: React.FC<IOATGraphCustomEdgeProps> = ({
     }, [id, source, sourceX, sourceY, targetX, targetY]);
 
     const onNameClick = () => {
-        setNameEditor(true);
+        const selectRelationship = () => {
+            setCurrentNode(polygons.element.id);
+            dispatch({
+                type: SET_OAT_SELECTED_MODEL_ID,
+                payload: polygons.element.data.id
+            });
+            setNameEditor(true);
+        };
 
-        const relationship = new DTDLRelationship(
-            polygons.element.data.id,
-            nameText,
-            polygons.element.data.displayName,
-            polygons.element.data.description,
-            polygons.element.data.comment,
-            polygons.element.data.writable,
-            polygons.element.data.content ? polygons.element.data.content : [],
-            polygons.element.data.target,
-            polygons.element.data.maxMultiplicity
+        execute(
+            () => selectRelationship(),
+            () => {
+                setCurrentNode(currentNodeIdRef.current);
+                dispatch({
+                    type: SET_OAT_SELECTED_MODEL_ID,
+                    payload: selectedModelId
+                });
+                setNameEditor(false);
```

Comment:
Let's wrap this inline function into a similarly named function as select relationship. I.e. unselectRelationship. We should do this for all the execute calls.